contributors/devel/architectural-roadmap.md

```diff
@@ -0,0 +1,1099 @@
+# Kubernetes Architectural Roadmap
+
+**Shared with the community**
+
+Status: First draft
+
+Last update: 4/20/2017
+
+Authors: Brian Grant, Tim Hockin, and Clayton Coleman
+
+Intended audience: Kubernetes contributors
+
+* * *
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [Kubernetes Architectural Roadmap](#kubernetes-architectural-roadmap)
+    - [Summary/TL;DR](#summarytldr)
+    - [Background](#background)
+    - [System Layers](#system-layers)
+        - [The Nucleus: API and Execution](#the-nucleus-api-and-execution)
+            - [The API and cluster control plane](#the-api-and-cluster-control-plane)
+            - [Execution](#execution)
+        - [The Application Layer: Deployment and Routing](#the-application-layer-deployment-and-routing)
+        - [The Governance Layer: Automation and Policy Enforcement](#the-governance-layer-automation-and-policy-enforcement)
+        - [The Interface Layer: Libraries and Tools](#the-interface-layer-libraries-and-tools)
+        - [The Ecosystem](#the-ecosystem)
+    - [Managing the matrix](#managing-the-matrix)
+    - [Layering of the system as it relates to security](#layering-of-the-system-as-it-relates-to-security)
+    - [Next Steps](#next-steps)
+
+<!-- markdown-toc end -->
+
+
+## Summary/TL;DR
+
+TBD - for those who want an overview, but don't want to read a 20 page document.
+
+## Background
+
+Kubernetes is a platform for deploying and managing containers. For
+more information about the mission, scope, and design of Kubernetes,
+see [What Is Kubernetes](http://kubernetes.io/docs/whatisk8s/) and the
+[architectural
+overview](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture.md). The
```

Comment:
Doesn't exist, is this in a PR?

Reply:
Good catch.  It has moved to architecture/architecture.md.  I'll fix the link.